core/os/device/deviceinfo/cc/query.cpp

```diff
@@ -72,36 +72,35 @@ void deviceInstanceID(device::Instance* instance) {
   delete[] proto_data;
 }
 
-void buildDeviceInstance(const query::Option& opt, device::Instance** out) {
+}  // anonymous namespace
+
+namespace query {
+
+device::Instance* getDeviceInstance(const Option& opt, std::string* error) {
   using namespace device;
   using namespace google::protobuf::io;
 
-  if (!query::createContext()) {
-    return;
+  PlatformInfo osInfo;
+  CpuInfo cpuInfo;
+  if (!query::queryPlatform(&osInfo, error) ||
+      !query::queryCpu(&cpuInfo, error)) {
+    return nullptr;
   }
 
   // OS
   auto os = new OS();
-  os->set_kind(query::osKind());
-  os->set_name(query::osName());
-  os->set_build(query::osBuild());
-  os->set_major_version(query::osMajor());
-  os->set_minor_version(query::osMinor());
-  os->set_point_version(query::osPoint());
+  os->set_kind(osInfo.osKind);
+  os->set_name(osInfo.osName);
+  os->set_build(osInfo.osBuild);
+  os->set_major_version(osInfo.osMajor);
+  os->set_minor_version(osInfo.osMinor);
+  os->set_point_version(osInfo.osPoint);
 
   // Instance.Configuration.Drivers
   auto drivers = new Drivers();
 
-  const char* backupVendor = "";
-  const char* backupName = "";
-  if (query::hasGLorGLES()) {
-    // Instance.Configuration.Drivers.OpenGLDriver
-    auto opengl_driver = new OpenGLDriver();
-    query::glDriver(opengl_driver);
-    drivers->set_allocated_opengl(opengl_driver);
-    backupVendor = opengl_driver->vendor().c_str();
-    backupName = opengl_driver->renderer().c_str();
-  }
+  const char* gpuVendor = "";  // TODO: we no longer have a vlue for this
```

Comment:
typo: vlue -> value

Is this something we should get via Vulkan now? (doesn't have to be in this PR)

Reply:
I've pushed another commit with a tiny PCI ID database to lookup the vendor name.